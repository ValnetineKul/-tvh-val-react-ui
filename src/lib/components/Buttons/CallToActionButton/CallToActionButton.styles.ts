import { makeStyles } from '../../../../themes/core';

type Props = {
  withIcon: boolean;
};

const useStyles = makeStyles<Props>({ name: 'CallToActionButton' })((theme, { withIcon }) => ({
  root: {
    transform: 'skewX(-10deg)',
    minHeight: 40,
    display: 'inline-flex',
    alignItems: 'center',
    // we need exactly 4px to compensate the skewX(10deg) with 40 height
    padding: `${theme.layoutSpacing['spacing/300']} ${theme.layoutSpacing['spacing/200']}`,
    '&:disabled': {
      backgroundColor: theme.color['bg/disabled'],
      color: theme.color['text/onSurfaceCurrent/disabled'],
    },
  },
  inner: {
    transform: 'skewX(10deg)',
    display: 'flex',
  },
  content: {
    display: 'flex',
    alignItems: 'center',
    padding: `${theme.layoutSpacing['spacing/000']} ${theme.layoutSpacing['spacing/400']}`,
  },
  iconOnly: {
    padding: `${theme.layoutSpacing['spacing/000']} ${theme.layoutSpacing['spacing/300']}`,
  },
  label: {
    marginLeft: withIcon ? theme.layoutSpacing['spacing/300'] : theme.layoutSpacing['spacing/000'],
  },
}));

export default useStyles;
