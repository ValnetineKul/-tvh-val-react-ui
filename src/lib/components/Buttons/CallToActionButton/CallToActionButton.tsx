import type { MouseEventHandler, ReactElement } from 'react';
import React, { forwardRef } from 'react';
import SpinnerWithBackdrop from '../../ProgressIndicators/Spinner/SpinnerWithBackdrop';
import ButtonBase from '../../Surfaces/ButtonBase';
import type { ButtonBaseProps } from '../../ButtonBase';
import Typography from '../../Typography';
import useStyles from './CallToActionButton.styles';
import useGetButtonProps from '../Button/helpers/useGetButtonProps';
import type { DataAttributes } from '../../../types/common';

export interface BasicProps extends ButtonBaseProps {
  icon?: ReactElement;
  label?: string;
  className?: string;
  variant?: 'primary' | 'secondary';
  isLoading?: boolean;
  onClick?: MouseEventHandler<HTMLButtonElement>;
  buttonProps?: React.HTMLAttributes<HTMLButtonElement | HTMLAnchorElement> & DataAttributes;
}

export type OptionalLabel = { label?: string; icon: ReactElement };
export type OptionalIcon = { label: string; icon?: ReactElement };

export type CallToActionButtonProps = (OptionalLabel | OptionalIcon) & BasicProps;

const COLORS = {
  primary: 'Primary',
  secondary: 'Secondary',
} as const;

const CallToActionButton = forwardRef<HTMLButtonElement, CallToActionButtonProps>(
  ({ icon, label, variant = 'primary', isLoading, className, ...props }, ref) => {
    const { classes, cx } = useStyles({ withIcon: !!icon });

    const { shouldShowLoader, buttonBaseProps } = useGetButtonProps(isLoading, props);

    return (
      <ButtonBase
        ref={ref}
        {...props.buttonProps}
        color={COLORS[variant]}
        className={cx(classes.root, className)}
        onClick={props.onClick}
        {...buttonBaseProps}
      >
        <div className={classes.inner}>
          <SpinnerWithBackdrop isLoading={shouldShowLoader} isBackdropOpaque>
            <div className={cx(classes.content, icon && !label && classes.iconOnly)}>
              {icon && React.cloneElement(icon, { size: 'md' })}
              {label && (
                <Typography component="span" variant="h4" headerType="commercial" className={classes.label}>
                  {label}
                </Typography>
              )}
            </div>
          </SpinnerWithBackdrop>
        </div>
      </ButtonBase>
    );
  }
);
CallToActionButton.displayName = 'CallToActionButton';

export default CallToActionButton;
