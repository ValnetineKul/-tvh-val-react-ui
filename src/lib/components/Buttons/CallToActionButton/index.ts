import CallToActionButton from './CallToActionButton';

export type { CallToActionButtonProps } from './CallToActionButton';
export default CallToActionButton;
