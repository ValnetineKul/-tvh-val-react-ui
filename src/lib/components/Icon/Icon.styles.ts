import { makeStyles } from '../../../themes/core';

const useStyles = makeStyles({ name: 'Icon' })(() => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  sizeXl: {
    fontSize: '64px !important',
    width: '80px',
    height: '80px',
    '& svg': {
      width: '80px',
      height: '80px',
    },
  },
  sizeMd: {
    fontSize: '19px !important',
    width: '24px',
    height: '24px',
    '& svg': {
      width: '24px',
      height: '24px',
    },
  },
  sizeSm: {
    fontSize: '12px !important',
    width: '16px',
    height: '16px',
    '& svg': {
      width: '16px',
      height: '16px',
    },
  },
}));

export default useStyles;
