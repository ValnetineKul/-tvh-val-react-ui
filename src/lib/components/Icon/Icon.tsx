import React from 'react';
import SvgIcon from '@mui/material/SvgIcon';
import { capitalize } from '@mui/material/utils';
import useStyles from './Icon.styles';

export interface IconProps extends React.AriaAttributes {
  icon: React.FunctionComponent;
  size?: 'md' | 'sm' | 'xl';
  className?: string;
  title?: string;
  isMultiColor?: boolean;
}

const Icon = ({ icon, size = 'md', className, title, isMultiColor = false, ...props }: IconProps) => {
  const { classes, cx } = useStyles();
  const sizeClassName = (classes as Record<string, string>)[`size${capitalize(size)}`];

  const IconComponent = typeof icon === 'string' ? React.lazy(() => import(`./icons/${icon}`)) : icon;

  return (
    <div className={cx(classes.root, sizeClassName, className)}>
      <SvgIcon
        htmlColor={isMultiColor ? 'transparent' : 'inherit'}
        component={IconComponent as React.ElementType}
        // a11y mui issue: combination "component + titleAccess" does not work: https://github.com/mui/material-ui/issues/33421
        titleAccess={title}
        {...props}
      />
    </div>
  );
};

export default Icon;
