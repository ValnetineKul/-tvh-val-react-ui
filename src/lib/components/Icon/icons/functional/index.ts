export { ReactComponent as AddressBook } from './svg/addressBook.svg';
export { ReactComponent as Analytics } from './svg/analytics.svg';
export { ReactComponent as AngleDown } from './svg/angleDown.svg';
export { ReactComponent as AngleLeft } from './svg/angleLeft.svg';
export { ReactComponent as AngleRight } from './svg/angleRight.svg';
export { ReactComponent as AngleUp } from './svg/angleUp.svg';
export { ReactComponent as ArrowAltCircleLeft } from './svg/arrowAltCircleLeft.svg';
export { ReactComponent as ArrowDown } from './svg/arrowDown.svg';
export { ReactComponent as ArrowDownArrowUp } from './svg/arrowDownArrowUp.svg';
export { ReactComponent as ArrowLeft } from './svg/arrowLeft.svg';
export { ReactComponent as ArrowLeftToLine } from './svg/arrowLeftToLine.svg';
export { ReactComponent as ArrowRight } from './svg/arrowRight.svg';
export { ReactComponent as ArrowRightArrowLeft } from './svg/arrowRightArrowLeft.svg';
export { ReactComponent as ArrowRightToLine } from './svg/arrowRightToLine.svg';
export { ReactComponent as ArrowUp } from './svg/arrowUp.svg';
export { ReactComponent as ArrowUpRightFromSquare } from './svg/arrowUpRightFromSquare.svg';
export { ReactComponent as Atlas } from './svg/atlas.svg';
export { ReactComponent as Award } from './svg/award.svg';
export { ReactComponent as Badge } from './svg/badge.svg';
export { ReactComponent as BadgePercent } from './svg/badgePercent.svg';
export { ReactComponent as BallotCheck } from './svg/ballotCheck.svg';
export { ReactComponent as Bars } from './svg/bars.svg';
export { ReactComponent as BatteryFull } from './svg/batteryFull.svg';
export { ReactComponent as BatteryHalf } from './svg/batteryHalf.svg';
export { ReactComponent as BatteryLow } from './svg/batteryLow.svg';
export { ReactComponent as Bell } from './svg/bell.svg';
export { ReactComponent as Book } from './svg/book.svg';
export { ReactComponent as BookAlt } from './svg/bookAlt.svg';
export { ReactComponent as BookmarkCheck } from './svg/bookmarkCheck.svg';
export { ReactComponent as BookOpen } from './svg/bookOpen.svg';
export { ReactComponent as Box } from './svg/box.svg';
export { ReactComponent as BoxAlt } from './svg/boxAlt.svg';
export { ReactComponent as Boxes } from './svg/boxes.svg';
export { ReactComponent as BoxFull } from './svg/boxFull.svg';
export { ReactComponent as BoxHeart } from './svg/boxHeart.svg';
export { ReactComponent as BoxOpen } from './svg/boxOpen.svg';
export { ReactComponent as Briefcase } from './svg/briefcase.svg';
export { ReactComponent as Building } from './svg/building.svg';
export { ReactComponent as BulkyItem } from './svg/bulkyItem.svg';
export { ReactComponent as BullseyeArrow } from './svg/bullseyeArrow.svg';
export { ReactComponent as BullseyePointer } from './svg/bullseyePointer.svg';
export { ReactComponent as Calculator } from './svg/calculator.svg';
export { ReactComponent as Calendar } from './svg/calendar.svg';
export { ReactComponent as CalendarCheck } from './svg/calendarCheck.svg';
export { ReactComponent as CalendarStar } from './svg/calendarStar.svg';
export { ReactComponent as Camera } from './svg/camera.svg';
export { ReactComponent as CarMechanic } from './svg/carMechanic.svg';
export { ReactComponent as CartArrowDown } from './svg/cartArrowDown.svg';
export { ReactComponent as Cat } from './svg/cat.svg';
export { ReactComponent as ChartLine } from './svg/chartLine.svg';
export { ReactComponent as ChartPie } from './svg/chartPie.svg';
export { ReactComponent as Check } from './svg/check.svg';
export { ReactComponent as CheckCircle } from './svg/checkCircle.svg';
export { ReactComponent as ChessKnight } from './svg/chessKnight.svg';
export { ReactComponent as ClipboardListCheck } from './svg/clipboardListCheck.svg';
export { ReactComponent as Clock } from './svg/clock.svg';
export { ReactComponent as Cog } from './svg/cog.svg';
export { ReactComponent as Cogs } from './svg/cogs.svg';
export { ReactComponent as Coin } from './svg/coin.svg';
export { ReactComponent as Coins } from './svg/coins.svg';
export { ReactComponent as Comment } from './svg/comment.svg';
export { ReactComponent as CommentsAlt } from './svg/commentsAlt.svg';
export { ReactComponent as ContainerStorage } from './svg/containerStorage.svg';
export { ReactComponent as ConveyorBelt } from './svg/conveyorBelt.svg';
export { ReactComponent as ConveyorBeltSlash } from './svg/conveyorBeltSlash.svg';
export { ReactComponent as Copy } from './svg/copy.svg';
export { ReactComponent as CreditCard } from './svg/creditCard.svg';
export { ReactComponent as DesktopAlt } from './svg/desktopAlt.svg';
export { ReactComponent as Dolly } from './svg/dolly.svg';
export { ReactComponent as DollyFlatbed } from './svg/dollyFlatbed.svg';
export { ReactComponent as DollyFlatbedAlt } from './svg/dollyFlatbedAlt.svg';
export { ReactComponent as Download } from './svg/download.svg';
export { ReactComponent as EllipsisH } from './svg/ellipsisH.svg';
export { ReactComponent as Envelope } from './svg/envelope.svg';
export { ReactComponent as ExclamationCircle } from './svg/exclamationCircle.svg';
export { ReactComponent as ExclamationTriangle } from './svg/exclamationTriangle.svg';
export { ReactComponent as Eye } from './svg/eye.svg';
export { ReactComponent as EyeSlash } from './svg/eyeSlash.svg';
export { ReactComponent as Fast } from './svg/fast.svg';
export { ReactComponent as FileAlt } from './svg/fileAlt.svg';
export { ReactComponent as FileCheck } from './svg/fileCheck.svg';
export { ReactComponent as FileDownload } from './svg/fileDownload.svg';
export { ReactComponent as FileInvoice } from './svg/fileInvoice.svg';
export { ReactComponent as FileUpload } from './svg/fileUpload.svg';
export { ReactComponent as Filter } from './svg/filter.svg';
export { ReactComponent as FlagAlt } from './svg/flagAlt.svg';
export { ReactComponent as Folder } from './svg/folder.svg';
export { ReactComponent as Forklift } from './svg/forklift.svg';
export { ReactComponent as Fragile } from './svg/fragile.svg';
export { ReactComponent as Gauge } from './svg/gauge.svg';
export { ReactComponent as GaugeMax } from './svg/gaugeMax.svg';
export { ReactComponent as GaugeMin } from './svg/gaugeMin.svg';
export { ReactComponent as Gift } from './svg/gift.svg';
export { ReactComponent as GiftCard } from './svg/giftCard.svg';
export { ReactComponent as Gifts } from './svg/gifts.svg';
export { ReactComponent as Globe } from './svg/globe.svg';
export { ReactComponent as GlobeAfrica } from './svg/globeAfrica.svg';
export { ReactComponent as GlobeAmericas } from './svg/globeAmericas.svg';
export { ReactComponent as GlobeAsia } from './svg/globeAsia.svg';
export { ReactComponent as GlobeEurope } from './svg/globeEurope.svg';
export { ReactComponent as GraduationCap } from './svg/graduationCap.svg';
export { ReactComponent as Grid } from './svg/grid.svg';
export { ReactComponent as HandHoldingBox } from './svg/handHoldingBox.svg';
export { ReactComponent as HandPointerSwipe } from './svg/handPointerSwipe.svg';
export { ReactComponent as Handshake } from './svg/handshake.svg';
export { ReactComponent as Headset } from './svg/headset.svg';
export { ReactComponent as HelmetSafety } from './svg/helmetSafety.svg';
export { ReactComponent as Hiking } from './svg/hiking.svg';
export { ReactComponent as History } from './svg/history.svg';
export { ReactComponent as Home } from './svg/home.svg';
export { ReactComponent as HorizontalRule } from './svg/horizontalRule.svg';
export { ReactComponent as Image } from './svg/image.svg';
export { ReactComponent as Inbox } from './svg/inbox.svg';
export { ReactComponent as Indent } from './svg/indent.svg';
export { ReactComponent as Infinite } from './svg/infinite.svg';
export { ReactComponent as InfoCircle } from './svg/infoCircle.svg';
export { ReactComponent as Inventory } from './svg/inventory.svg';
export { ReactComponent as InvoiceOverdue } from './svg/invoiceOverdue.svg';
export { ReactComponent as Language } from './svg/language.svg';
export { ReactComponent as Leaf } from './svg/leaf.svg';
export { ReactComponent as LevelRight } from './svg/levelRight.svg';
export { ReactComponent as LightbulbOn } from './svg/lightbulbOn.svg';
export { ReactComponent as Link } from './svg/link.svg';
export { ReactComponent as LinkSlash } from './svg/linkSlash.svg';
export { ReactComponent as List } from './svg/list.svg';
export { ReactComponent as Lock } from './svg/lock.svg';
export { ReactComponent as LockOpen } from './svg/lockOpen.svg';
export { ReactComponent as MapLocationDot } from './svg/mapLocationDot.svg';
export { ReactComponent as MapMarkerAlt } from './svg/mapMarkerAlt.svg';
export { ReactComponent as Medal } from './svg/medal.svg';
export { ReactComponent as Megaphone } from './svg/megaphone.svg';
export { ReactComponent as Microchip } from './svg/microchip.svg';
export { ReactComponent as Minus } from './svg/minus.svg';
export { ReactComponent as MobileButton } from './svg/mobileButton.svg';
export { ReactComponent as MoneyBillWave } from './svg/moneyBillWave.svg';
export { ReactComponent as MoneyCheckEdit } from './svg/moneyCheckEdit.svg';
export { ReactComponent as Narwhal } from './svg/narwhal.svg';
export { ReactComponent as NonReturnableItem } from './svg/nonReturnableItem.svg';
export { ReactComponent as Outdent } from './svg/outdent.svg';
export { ReactComponent as Pallet } from './svg/pallet.svg';
export { ReactComponent as Paperclip } from './svg/paperclip.svg';
export { ReactComponent as PaperPlane } from './svg/paperPlane.svg';
export { ReactComponent as ParachuteBox } from './svg/parachuteBox.svg';
export { ReactComponent as Pen } from './svg/pen.svg';
export { ReactComponent as PeopleArrowsLeftRight } from './svg/peopleArrowsLeftRight.svg';
export { ReactComponent as PeopleGroup } from './svg/peopleGroup.svg';
export { ReactComponent as PersonDolly } from './svg/personDolly.svg';
export { ReactComponent as PersonRays } from './svg/personRays.svg';
export { ReactComponent as Phone } from './svg/phone.svg';
export { ReactComponent as Plane } from './svg/plane.svg';
export { ReactComponent as PlaneSlash } from './svg/planeSlash.svg';
export { ReactComponent as Play } from './svg/play.svg';
export { ReactComponent as PlayCircle } from './svg/playCircle.svg';
export { ReactComponent as Plus } from './svg/plus.svg';
export { ReactComponent as Print } from './svg/print.svg';
export { ReactComponent as PuzzlePiece } from './svg/puzzlePiece.svg';
export { ReactComponent as QrCode } from './svg/qrCode.svg';
export { ReactComponent as QuestionCircle } from './svg/questionCircle.svg';
export { ReactComponent as Random } from './svg/random.svg';
export { ReactComponent as Recycle } from './svg/recycle.svg';
export { ReactComponent as Redo } from './svg/redo.svg';
export { ReactComponent as ResearchRequest } from './svg/researchRequest.svg';
export { ReactComponent as ReturnableItem } from './svg/returnableItem.svg';
export { ReactComponent as Ribbon } from './svg/ribbon.svg';
export { ReactComponent as RocketLaunch } from './svg/rocketLaunch.svg';
export { ReactComponent as Route } from './svg/route.svg';
export { ReactComponent as RulerHorizontal } from './svg/rulerHorizontal.svg';
export { ReactComponent as Scanner } from './svg/scanner.svg';
export { ReactComponent as Screwdriver } from './svg/screwdriver.svg';
export { ReactComponent as Search } from './svg/search.svg';
export { ReactComponent as SearchMinus } from './svg/searchMinus.svg';
export { ReactComponent as SearchPlus } from './svg/searchPlus.svg';
export { ReactComponent as ShieldCheck } from './svg/shieldCheck.svg';
export { ReactComponent as Ship } from './svg/ship.svg';
export { ReactComponent as ShippingFast } from './svg/shippingFast.svg';
export { ReactComponent as ShippingTimed } from './svg/shippingTimed.svg';
export { ReactComponent as ShoppingBasket } from './svg/shoppingBasket.svg';
export { ReactComponent as ShoppingCart } from './svg/shoppingCart.svg';
export { ReactComponent as SignOut } from './svg/signOut.svg';
export { ReactComponent as Smile } from './svg/smile.svg';
export { ReactComponent as Snowplow } from './svg/snowplow.svg';
export { ReactComponent as SolarPanel } from './svg/solarPanel.svg';
export { ReactComponent as Sparkles } from './svg/sparkles.svg';
export { ReactComponent as Star } from './svg/star.svg';
export { ReactComponent as Stopwatch } from './svg/stopwatch.svg';
export { ReactComponent as Sync } from './svg/sync.svg';
export { ReactComponent as Tag } from './svg/tag.svg';
export { ReactComponent as Tags } from './svg/tags.svg';
export { ReactComponent as ThumbsUp } from './svg/thumbsUp.svg';
export { ReactComponent as Ticket } from './svg/ticket.svg';
export { ReactComponent as TimelineArrow } from './svg/timelineArrow.svg';
export { ReactComponent as Times } from './svg/times.svg';
export { ReactComponent as TimesCircle } from './svg/timesCircle.svg';
export { ReactComponent as Tools } from './svg/tools.svg';
export { ReactComponent as Tractor } from './svg/tractor.svg';
export { ReactComponent as Trash } from './svg/trash.svg';
export { ReactComponent as Trophy } from './svg/trophy.svg';
export { ReactComponent as TrophyAlt } from './svg/trophyAlt.svg';
export { ReactComponent as Truck } from './svg/truck.svg';
export { ReactComponent as TruckContainer } from './svg/truckContainer.svg';
export { ReactComponent as TruckFree } from './svg/truckFree.svg';
export { ReactComponent as TruckFreeSlash } from './svg/truckFreeSlash.svg';
export { ReactComponent as TruckLoading } from './svg/truckLoading.svg';
export { ReactComponent as TruckPickup } from './svg/truckPickup.svg';
export { ReactComponent as TruckPlow } from './svg/truckPlow.svg';
export { ReactComponent as UnknownPart } from './svg/unknownPart.svg';
export { ReactComponent as Unlink } from './svg/unlink.svg';
export { ReactComponent as UpRightAndDownLeftFromCenter } from './svg/upRightAndDownLeftFromCenter.svg';
export { ReactComponent as User } from './svg/user.svg';
export { ReactComponent as UserCircle } from './svg/userCircle.svg';
export { ReactComponent as UserHeadset } from './svg/userHeadset.svg';
export { ReactComponent as Users } from './svg/users.svg';
export { ReactComponent as UserShield } from './svg/userShield.svg';
export { ReactComponent as Video } from './svg/video.svg';
export { ReactComponent as VideoSlash } from './svg/videoSlash.svg';
export { ReactComponent as View360 } from './svg/view360.svg';
export { ReactComponent as Wallet } from './svg/wallet.svg';
export { ReactComponent as WalletCheck } from './svg/walletCheck.svg';
export { ReactComponent as WarehouseFull } from './svg/warehouseFull.svg';
export { ReactComponent as WeightHanging } from './svg/weightHanging.svg';
export { ReactComponent as Wheat } from './svg/wheat.svg';
export { ReactComponent as Wifi } from './svg/wifi.svg';
export { ReactComponent as WifiSlash } from './svg/wifiSlash.svg';
export { ReactComponent as Wrench } from './svg/wrench.svg';
