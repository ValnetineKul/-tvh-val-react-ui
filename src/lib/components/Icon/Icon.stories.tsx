import type { FunctionComponent, SVGProps } from 'react';
import React from 'react';
import Typography from '../Typography';
import { createMeta } from '../../story-utils';
import Icon from './Icon';
import * as countryIcons from './icons/country';
import * as functionalIcons from './icons/functional';
import * as productCategory from './icons/productCategory';
import * as machinery from './icons/machinery';
import * as commercial from './icons/commercial';
import * as brand from './icons/brand';

type Size = 'xl' | 'md' | 'sm';

const renderIconsListWithLetter = <
  T extends Record<string, FunctionComponent<SVGProps<SVGSVGElement> & { title?: string | undefined }>>
>(
  icons: T,
  size: Size,
  isMultiColor = false
) => {
  const iconsNames: string[] = Object.keys(icons);

  iconsNames.sort((a: string, b: string) => a.localeCompare(b));

  let letter = '';
  return iconsNames.map((name) => {
    if (name[0] === letter) {
      return null;
    }
    [letter] = name;

    return (
      <div
        style={{
          display: 'flex',
          flexWrap: 'wrap',
          alignItems: 'center',
          padding: 10,
          marginBottom: 20,
          border: '1px solid black',
          borderRadius: 12,
        }}
        key={name}
      >
        <div style={{ marginRight: 20 }}>
          <Typography variant="h1" headerType="commercial">
            {name[0]}
          </Typography>
        </div>

        {iconsNames.map((iconName) => {
          if (iconName[0] !== letter) {
            return null;
          }

          return (
            <div style={{ padding: 10, display: 'flex', flexDirection: 'column', alignItems: 'center' }} key={iconName}>
              <Icon isMultiColor={isMultiColor} icon={icons[iconName as keyof T]} size={size} />

              <Typography>{iconName}</Typography>
            </div>
          );
        })}
      </div>
    );
  });
};

const FunctionalIconsListTemplate = ({ size }: { size: Size }) => {
  return (
    <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(functionalIcons, size)}</div>
  );
};

export const FunctionalIconsList = FunctionalIconsListTemplate.bind({});
FunctionalIconsList.args = {
  size: 'md',
};
FunctionalIconsList.storyName = 'Functional icons list';

const CountryIconsListTemplate = ({ size }: { size: Size }) => {
  return (
    <div style={{ display: 'flex', flexDirection: 'column' }}>
      {renderIconsListWithLetter(countryIcons, size, true)}
    </div>
  );
};

export const CountryIconsList = CountryIconsListTemplate.bind({});
CountryIconsList.args = {
  size: 'md',
};
CountryIconsList.storyName = 'Country icons list';

const ProductCategoryIconsListTemplate = ({ size }: { size: Size }) => {
  return (
    <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(productCategory, size)}</div>
  );
};

export const ProductCategoryIconsList = ProductCategoryIconsListTemplate.bind({});
ProductCategoryIconsList.args = {
  size: 'md',
};
ProductCategoryIconsList.storyName = 'Product category icons list';

const MachineryIconsListTemplate = ({ size }: { size: Size }) => {
  return <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(machinery, size)}</div>;
};

export const MachineryIconsList = MachineryIconsListTemplate.bind({});
MachineryIconsList.args = {
  size: 'md',
};
MachineryIconsList.storyName = 'Machinery icons list';

const CommercialIconsListTemplate = ({ size }: { size: Size }) => {
  return <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(commercial, size)}</div>;
};

export const CommercialIconsList = CommercialIconsListTemplate.bind({});
CommercialIconsList.args = {
  size: 'md',
};
CommercialIconsList.storyName = 'Commercial icons list';

const BrandIconsListTemplate = ({ size }: { size: Size }) => {
  return <div style={{ display: 'flex', flexDirection: 'column' }}>{renderIconsListWithLetter(brand, size)}</div>;
};

export const BrandIconsList = BrandIconsListTemplate.bind({});
BrandIconsList.args = {
  size: 'md',
};
BrandIconsList.storyName = 'Brand icons list';

export default createMeta({
  component: Icon,
  title: 'Foundations/Icon',
  argTypes: {
    icon: {
      control: {
        disable: true,
      },
      table: {
        type: { summary: 'FontAwesome icon definition' },
      },
    },
    isMultiColor: {
      control: {
        disable: true,
      },
    },
    title: {
      control: {
        disable: true,
      },
    },
  },
});
