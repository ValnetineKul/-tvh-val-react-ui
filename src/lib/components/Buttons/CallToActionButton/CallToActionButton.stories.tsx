import React from 'react';
import { action } from '@storybook/addon-actions';
import CallToActionButton from './CallToActionButton';
import { createMeta, createTemplate } from '../../../story-utils';
import { AngleLeft } from '../../Icon/icons/functional';
import Icon from '../../Icon';
import { buttonBaseStoryArgs } from '../../ButtonBase';

const Template = createTemplate(CallToActionButton);

export const Primitive = Template.bind({});
Primitive.args = {
  onClick: action('onClick'),
  label: 'Label',
  variant: 'primary',
  buttonProps: {
    'data-tms-banner-name': 'title',
    'data-tms-banner-id': 'drupal_internal__id',
    'data-tms-banner-creative': `sectionLabel field_banner_size`,
  },
};

Primitive.storyName = 'CallToActionButton';

export const WithIcon = Template.bind({});
WithIcon.args = {
  onClick: action('onClick'),
  icon: <Icon icon={AngleLeft} />,
  label: 'Label',
  variant: 'primary',
};

WithIcon.storyName = 'startIcon';

export const WithoutLabel = Template.bind({});
WithoutLabel.args = {
  onClick: action('onClick'),
  icon: <Icon icon={AngleLeft} />,
  variant: 'primary',
};

WithoutLabel.storyName = 'iconOnly';

export default createMeta({
  component: CallToActionButton,
  title: 'Components/Buttons/CallToActionButton',
  argTypes: {
    onClick: {
      control: { disable: true },
    },
    icon: {
      control: {
        disable: true,
      },
      table: {
        type: { summary: 'Icon with FontAwesome definition' },
      },
    },
    ...buttonBaseStoryArgs,
  },
});
