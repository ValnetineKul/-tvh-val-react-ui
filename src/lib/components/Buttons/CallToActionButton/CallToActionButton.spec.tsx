import React from 'react';
import { Search } from '../../Icon/icons/functional';
import CallToActionButton from './CallToActionButton';
import Icon from '../../Icon';
import { render, userEvent } from '../../../test-utils';

describe('CallToActionButton', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <CallToActionButton onClick={jest.fn()} label="Label" icon={<Icon icon={Search} />} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should render correctly without label', () => {
    const { container } = render(<CallToActionButton onClick={jest.fn()} icon={<Icon icon={Search} />} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should render correctly with label', () => {
    const { container } = render(<CallToActionButton onClick={jest.fn()} label="Label" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should correctly render a link', () => {
    const { container, getByRole } = render(<CallToActionButton href="#" label="Label" />);
    expect(container.firstChild).toMatchSnapshot();

    expect(getByRole('link')).toBeInTheDocument();
  });

  it('Should set disabled property', () => {
    const { container } = render(<CallToActionButton onClick={jest.fn()} disabled label="Label" />);
    expect(container.firstChild as HTMLElement).toBeDisabled();
  });

  it('Should be clickable and invoke callback function', () => {
    const handleClick = jest.fn();
    const { getByRole } = render(<CallToActionButton onClick={handleClick} icon={<Icon icon={Search} />} />);
    const button = getByRole('button');
    userEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('Should render spinner', () => {
    const { getByRole } = render(<CallToActionButton onClick={jest.fn()} label="Label" isLoading />);
    expect(getByRole('progressbar')).toBeInTheDocument();
  });
});
