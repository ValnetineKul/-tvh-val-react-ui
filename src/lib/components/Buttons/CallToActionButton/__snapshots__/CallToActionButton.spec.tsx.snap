// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToActionButton Should correctly render a link 1`] = `
<a
  aria-disabled="false"
  class="MuiButtonBase-root tss-1wztk4a-ButtonBase-root-ButtonBase-focusOut-CallToActionButton-root mui-10d1a0h-MuiButtonBase-root"
  href="#"
  tabindex="0"
>
  <div
    class="tss-ddcb3u-CallToActionButton-inner"
  >
    <div
      class="tss-pyz8vm-SpinnerWithBackdrop-root"
    >
      <div
        class=""
      >
        <div
          class="tss-142dt2m-CallToActionButton-content"
        >
          <span
            class="tss-3-Typography-headerCommercial-ref tss-1uop2om-Typography-root-Typography-headerCommercial-CallToActionButton-label-Typography-h4"
          >
            Label
          </span>
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`CallToActionButton Should render correctly 1`] = `
<button
  class="MuiButtonBase-root tss-1wztk4a-ButtonBase-root-ButtonBase-focusOut-CallToActionButton-root mui-10d1a0h-MuiButtonBase-root"
  tabindex="0"
  type="button"
>
  <div
    class="tss-ddcb3u-CallToActionButton-inner"
  >
    <div
      class="tss-pyz8vm-SpinnerWithBackdrop-root"
    >
      <div
        class=""
      >
        <div
          class="tss-142dt2m-CallToActionButton-content"
        >
          <div
            class="tss-w1sjks-Icon-root-Icon-sizeMd"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
              color="inherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              search.svg
            </svg>
          </div>
          <span
            class="tss-3-Typography-headerCommercial-ref tss-pfp3h7-Typography-root-Typography-headerCommercial-CallToActionButton-label-Typography-h4"
          >
            Label
          </span>
        </div>
      </div>
    </div>
  </div>
</button>
`;

exports[`CallToActionButton Should render correctly with label 1`] = `
<button
  class="MuiButtonBase-root tss-1wztk4a-ButtonBase-root-ButtonBase-focusOut-CallToActionButton-root mui-10d1a0h-MuiButtonBase-root"
  tabindex="0"
  type="button"
>
  <div
    class="tss-ddcb3u-CallToActionButton-inner"
  >
    <div
      class="tss-pyz8vm-SpinnerWithBackdrop-root"
    >
      <div
        class=""
      >
        <div
          class="tss-142dt2m-CallToActionButton-content"
        >
          <span
            class="tss-3-Typography-headerCommercial-ref tss-1uop2om-Typography-root-Typography-headerCommercial-CallToActionButton-label-Typography-h4"
          >
            Label
          </span>
        </div>
      </div>
    </div>
  </div>
</button>
`;

exports[`CallToActionButton Should render correctly without label 1`] = `
<button
  class="MuiButtonBase-root tss-1wztk4a-ButtonBase-root-ButtonBase-focusOut-CallToActionButton-root mui-10d1a0h-MuiButtonBase-root"
  tabindex="0"
  type="button"
>
  <div
    class="tss-ddcb3u-CallToActionButton-inner"
  >
    <div
      class="tss-pyz8vm-SpinnerWithBackdrop-root"
    >
      <div
        class=""
      >
        <div
          class="tss-1sqsqmd-CallToActionButton-content-CallToActionButton-iconOnly"
        >
          <div
            class="tss-w1sjks-Icon-root-Icon-sizeMd"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
              color="inherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              search.svg
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</button>
`;
